core/hash_map.h

```diff
@@ -74,13 +78,14 @@ class HashMapHahserDefault {
  * @param TKey  Key, search is based on it, needs to be hasheable. It is unique in this container.
  * @param TData Data, data associated with the key
  * @param Hasher Hasher object, needs to provide a valid static hash function for TKey
+ * @param Comparator comparator object, needs to be able to safely compare two TKey values
```

Comment:
> Go doesn't cleverly work around it, it doesn't IMPLEMENT the feature we're using. In go the variant equivalent could just not be used as a key in a hash at all.

That "feature" you're talking about is a workaround, it is something you invented. There is no such standard. Go just uses has a different workaround treating NaN float keys. You *can* add items with NaN values multiple times; instead of hashing them to the same bin like you do, they chose to put items with NaN keys in random bins.